cmd/soroban-rpc/internal/methods/get_ledgers.go

```diff
@@ -0,0 +1,243 @@
+		OldestLedgerCloseTime: ledgerRange.FirstLedger.CloseTime,
+		Cursor:                cursor,
+	}, nil
+}
+
+// initializePagination parses the request pagination details initializes the cursor.
+func (h ledgersHandler) initializePagination(request GetLedgersRequest,
+	ledgerRange ledgerbucketwindow.LedgerRange,
+) (uint32, uint, error) {
+	if request.Pagination == nil {
+		return request.StartLedger, h.defaultLimit, nil
+	}
+
+	start := request.StartLedger
+	var err error
+	if request.Pagination.Cursor != "" {
+		start, err = h.parseCursor(request.Pagination.Cursor, ledgerRange)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+
+	limit := request.Pagination.Limit
+	if limit <= 0 {
+		limit = h.defaultLimit
+	}
+	return start, limit, nil
+}
+
+func (h ledgersHandler) parseCursor(cursor string, ledgerRange ledgerbucketwindow.LedgerRange) (uint32, error) {
+	cursorInt, err := strconv.ParseUint(cursor, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+
+	start := uint32(cursorInt) + 1
+	if !isStartLedgerWithinBounds(start, ledgerRange) {
+		return 0, fmt.Errorf(
+			"cursor must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance",
+			ledgerRange.FirstLedger.Sequence,
+			ledgerRange.LastLedger.Sequence,
+		)
+	}
+
+	return start, nil
+}
+
+// fetchLedgers fetches ledgers from the DB for the range [start, start+limit-1]
+func (h ledgersHandler) fetchLedgers(ctx context.Context, start uint32,
+	limit uint, format string,
+) ([]LedgerInfo, error) {
+	ledgers, err := h.ledgerReader.BatchGetLedgers(ctx, start, limit)
+	if err != nil {
+		return nil, &jrpc2.Error{
+			Code:    jrpc2.InternalError,
+			Message: fmt.Sprintf("error fetching ledgers from db: %v", err),
+		}
+	}
+
+	result := make([]LedgerInfo, 0, limit)
+	for _, ledger := range ledgers {
+		if uint(len(result)) >= limit {
+			break
+		}
+
+		ledgerInfo, err := h.parseLedgerInfo(ledger, format)
+		if err != nil {
+			return nil, &jrpc2.Error{
+				Code:    jrpc2.InternalError,
+				Message: fmt.Sprintf("error processing ledger %d: %v", ledger.LedgerSequence(), err),
+			}
+		}
+		result = append(result, ledgerInfo)
+	}
+
+	return result, nil
+}
+
+// parseLedgerInfo extracts and formats the ledger metadata and header information.
+func (h ledgersHandler) parseLedgerInfo(ledger xdr.LedgerCloseMeta, format string) (LedgerInfo, error) {
+	ledgerInfo := LedgerInfo{
+		Hash:            ledger.LedgerHash().HexString(),
+		Sequence:        ledger.LedgerSequence(),
+		LedgerCloseTime: ledger.LedgerCloseTime(),
+	}
+	ledgerHeader := ledger.LedgerHeaderHistoryEntry()
+
+	// Format the data according to the requested format (JSON or XDR)
+	switch format {
+	case FormatJSON:
+		closeMetaJSON, headerJSON, convErr := ledgerToJSON(ledger, ledgerHeader)
+		if convErr != nil {
+			return LedgerInfo{}, convErr
+		}
+		ledgerInfo.LedgerCloseMetaJSON = closeMetaJSON
+		ledgerInfo.LedgerHeaderJSON = headerJSON
+	default:
+		closeMetaB, err := ledger.MarshalBinary()
```

Comment:
Isn't there a `MarshalBinaryBase64`?

Reply:
@Shaptic That method is part of `xdr.LedgerKey` and not the ledger metadata xdr